src/Provider.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from "react"
+import { shallowEqual } from "./utils/utils"
+import { IValueMap } from "./inject"
+
+export const MobXProviderContext: React.Context<any> = React.createContext({})
+
+export interface ProviderProps extends IValueMap {
+    children: React.ReactNode
+}
+
+export function Provider(props: ProviderProps): React.ReactElement {
+    const { children, ...stores } = props
+    const parentValue: any = React.useContext(MobXProviderContext)
+    const value: React.MutableRefObject<any> = React.useRef({
```

Comment:
Okay, no problem.  I don't understand what you mean by "should work fine" though, can you explain?

Reply:
I mean it should infer the correct type here. Although once you set type for `createContext`, it will complain here. You will need to declare an alias `type TProviderShape = Record<string, any>` and then `useRef<TProviderShape>`. Some other places too probably.

Reply:
I understand.   Thanks for clarifying.